python/audit/management/commands/clearaudit.py

```diff
@@ -140,11 +140,25 @@ def __prepare_csvs(self, *querysets, base_dir):
                 continue
 
             tmp_cvf_name = self.__get_csv_name(qs, now, base_dir)
-            with open(tmp_cvf_name, 'wt', newline='', encoding=self.encoding) as csv_file:
+            exists = os.path.exists(tmp_cvf_name)
+            mode = "at" if exists else "wt"
+            with open(tmp_cvf_name, mode, newline="", encoding=self.encoding) as csv_file:
                 writer = csv.writer(csv_file)
 
                 fields = [f.column for f in qs.model._meta.fields]
-                writer.writerow(fields)  # header
+                if not exists:
+                    writer.writerow(fields)  # header
+                else:
+                    with open(tmp_cvf_name, "rt", encoding=self.encoding) as csv_file:
```

Comment:
It's too complex to open file inside opened file context, maybe needs to be refactored

Reply:
refactored